aws_lambda_powertools/utilities/data_classes/dynamo_db_stream_event.py

```diff
@@ -73,6 +73,10 @@ def _deserialize_bool(self, value: bool) -> bool:
         return value
 
     def _deserialize_n(self, value: str) -> Decimal:
+        if len(value) > 38:
+            tail = len(value[38:]) - len(value[38:].rstrip("0"))
+            value = value[:-tail]
```

Comment:
If value is something like `00000000001234567891234567891234567834343978` this code will fail. Value is greater than 38, but there is no trailing zeros, what will result in the invalid expression `value = value[-0]`.

If I'm not wrong, Decimal always remove leading zeros, what means that we can use `lstrip` to remove leading zeros. Make sense?